src/options.js

```diff
@@ -198,51 +164,16 @@ function removeWhitelistDomain(event)
  * Gets all encountered origins with associated actions.
  * @return {Object}
  */
-function getOrigins()
-{
-  origins = {};
-
-  // Process origins allowed by user.
-  var green = FilterStorage.knownSubscriptions.userGreen.filters;
-  for (var i = 0; i < green.length; i++) {
-    origins[green[i].regexp.source] = 'usernoaction';
-  }
-
-  // Process origins cookie-blocked by user.
-  var yellow = FilterStorage.knownSubscriptions.userYellow.filters;
-  for (var i = 0; i < yellow.length; i++) {
-    origins[yellow[i].regexp.source] = 'usercookieblock';
-  }
-
-  // Process origins blocked by user.
-  var red = FilterStorage.knownSubscriptions.userRed.filters;
-  for (var i = 0; i < red.length; i++) {
-    origins[red[i].regexp.source] = 'userblock';
-  }
-
-  // Process origins blocked/cookie-blocked by heuristic.
-  var heuristics = FilterStorage.knownSubscriptions.frequencyHeuristic.filters;
-  for (var i = 0; i < heuristics.length; i++){
-    var origin = heuristics[i].regexp.source;
-    if (origins[origin]) {
-      continue;
-    }
-    if (filterWhitelist[origin]) {
-      origins[origin] = 'cookieblock';
-    } else {
-      origins[origin] = 'block';
-    }
-  }
-
-  // Process origins that have been seen but not blocked yet.
-  var seen = Object.keys(seenThirdParties);
-  for (var i = 0; i < seen.length; i++) {
-    var origin = seen[i];
-    if (! origins[origin]) {
-      origins[origin] = 'noaction';
-    }
+function getOrigins() {
+  var origins = {};
+  var action_map = pb.storage.getBadgerStorageObject('action_map');
+  for (var domain in action_map.getItemClones()) {
+      var action = pb.storage.getBestAction(domain);
+      // Do not show non tracking origins
+      if(action != pb.NO_TRACKING){
```

Comment:
Yea, I felt like it cluttered things too much? Although that may not be the right call. I am willing to consider changing this behavior.